sdk/python/feast/registry.py

```diff
@@ -54,31 +51,77 @@ class Registry:
         Create the Registry object.
 
         Args:
+            registry_config: RegistryConfig object containing the destination path and cache ttl,
             repo_path: Path to the base of the Feast repository
-            cache_ttl: The amount of time that cached registry state stays valid
-            registry_path: filepath or GCS URI that is the location of the object store registry,
             or where it will be created if it does not exist yet.
         """
+        registry_store_type = registry_config.registry_store_type
+        registry_path = registry_config.path
         uri = urlparse(registry_path)
-        if uri.scheme == "gs":
-            self._registry_store: RegistryStore = GCSRegistryStore(registry_path)
-        elif uri.scheme == "s3":
-            self._registry_store = S3RegistryStore(registry_path)
-        elif uri.scheme == "file" or uri.scheme == "":
-            self._registry_store = LocalRegistryStore(
-                repo_path=repo_path, registry_path_string=registry_path
-            )
+        if registry_store_type is None:
+            if uri.scheme == "gs":
+                from feast.infra.gcp import GCSRegistryStore
+
+                self._registry_store: RegistryStore = GCSRegistryStore(registry_path)
+            elif uri.scheme == "s3":
+                from feast.infra.aws import S3RegistryStore
+
+                self._registry_store = S3RegistryStore(registry_path)
+            elif uri.scheme == "file" or uri.scheme == "":
+                from feast.infra.local import LocalRegistryStore
+
+                self._registry_store = LocalRegistryStore(
+                    repo_path=repo_path, registry_path_string=registry_path
+                )
+            else:
+                raise Exception(
+                    f"Registry path {registry_path} has unsupported scheme {uri.scheme}. "
+                    f"Supported schemes are file, s3 and gs."
+                )
         else:
-            raise Exception(
-                f"Registry path {registry_path} has unsupported scheme {uri.scheme}. "
-                f"Supported schemes are file and gs."
-            )
-        self.cached_registry_proto_ttl = cache_ttl
+            registry_store_type = str(registry_store_type)
+            if "." not in registry_store_type:
+                if uri.scheme == "gs":
+                    from feast.infra.gcp import GCSRegistryStore
+
+                    self._registry_store = GCSRegistryStore(registry_path)
+                elif uri.scheme == "s3":
+                    from feast.infra.aws import S3RegistryStore
+
+                    self._registry_store = S3RegistryStore(registry_path)
+                elif uri.scheme == "file" or uri.scheme == "":
+                    from feast.infra.local import LocalRegistryStore
+
+                    self._registry_store = LocalRegistryStore(
+                        repo_path=repo_path, registry_path_string=registry_path
+                    )
```

Comment:
When would we hit this case? When users specified `registry_store_type: S3RegistryStore`, something like that?

I think that in that case, we probably want to have a dict of the "shortcut" types mapping to the fully qualified type (like in https://github.com/feast-dev/feast/blob/master/sdk/python/feast/repo_config.py#L22), and use the dynamic `importer.get_class_from_type` path.

Reply:
@achals thanks for the review.
I did the modifications, however, since there is a single constructor call `self._registry_store = cls(registry_config, repo_path)` this enforces `S3RegistryStore` and `GCSRegistryStore` constructors to have the signature:
`def __init__(self, registry_config: RegistryConfig, repo_path: Path):`
This is a bit awkward IMO since `repo_path` is not being used in those classes, it is just for the `LocalRegistryStore`
Let me know what you think if there is a better approach to consolidate those constructors.

Reply:
I think it's better to have a single interface even if it means passing in an unused parameter in some of the implementations, so I'd be okay with that change.

Reply:
thanks, @achals. The change is already in the latest modification on the PR.